adds a label for initContainers, test and user-guide

sosan opened the pull request:
#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
-->
/kind feature 
#### What this PR does / why we need it:
Add label for initContainers
for example: 
kompose.init-contaners.name: "init-service"
kompose.init-contaners.image: "busybox:1.28"
kompose.init-contaners.command: "['sh', '-c', 'until nslookup myservice.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for myservice; sleep 2; done']"

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Feature #1772

#### Special notes for your reviewer:

TessaIO commented on docs/user-guide.md:
````diff
@@ -467,6 +470,49 @@ services:
+For example:
+
+```yaml
+version: '2'
````
Let's refer to new versions
```suggestion
version: '3'
```

TessaIO commented on docs/user-guide.md:
````diff
@@ -467,6 +470,49 @@ services:
       kompose.volume.sub-path: pg-data
 ```
+
+
````
small nit here:
```suggestion
```

TessaIO commented on docs/user-guide.md:
````diff
@@ -467,6 +470,49 @@ services:
+For example:
+
+```yaml
+version: '2'
````
same as above

TessaIO commented on docs/user-guide.md:
````diff
@@ -467,6 +470,49 @@ services:
+For example:
+
+```yaml
+version: '2'
````
same as above

TessaIO commented on pkg/transformer/kubernetes/k8sutils.go:
```diff
@@ -985,3 +985,49 @@ func reformatSecretConfigUnderscoreWithDash(secretConfig types.ServiceSecretConf
 	return newSecretConfig
 }
+
+//	looks for an initContainer resources and its passed as labels
+//
+// if there is no image, it does not fill the initContainer
+// https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
```
Let's conform with the Golang comments style here. I suggest:
```suggestion
// fillInitContainers looks for an initContainer resources and its passed as labels
// if there is no image, it does not fill the initContainer
// https://kubernetes.io/docs/concepts/workloads/pods/init-containers/
```

TessaIO commented on pkg/transformer/kubernetes/k8sutils.go:
```diff
@@ -985,3 +985,49 @@ func reformatSecretConfigUnderscoreWithDash(secretConfig types.ServiceSecretConf
+	})
+}
+
+// parses a string containing comma-separated commands
+// returns a slice of strings or a single command and
+// example:
+// [ "bundle", "exec", "thin", "-p", "3000" ]
+//
+// example:
+// bundle exec thin -p 3000
```
Same as above:
```suggestion
// parseContainerCommandsFromStr parses a string containing comma-separated commands
// returns a slice of strings or a single command and
// example:
// [ "bundle", "exec", "thin", "-p", "3000" ]
//
// example:
// bundle exec thin -p 3000
```

TessaIO reviewed (changes requested):
@sosan Thanks for the PR. I put some comments below, and I'll do another round of review once my comments are addressed. One thing to add, can you add some e2e tests for this?

sosan commented:
> @sosan Thanks for the PR. I put some comments below, and I'll do another round of review once my comments are addressed. One thing to add, can you add some e2e tests for this?

suggestions fixed
added test e2e for initcontainer

```
===> Starting test <===
convert::expect_success_and_warning: Running: 'kompose -f /home/runner/work/kompose/kompose/script/test/fixtures/initcontainer/compose.yaml convert --stdout --with-kompose-annotation=false' expected_output: '/home/runner/work/kompose/kompose/script/test/fixtures/initcontainer/output-k8s.yaml' expected_warning: ''
PASS: converted output matches
PASS: warning found: ''
```

cdrage reviewed (changes requested):
Sorry for the delay!

Some feedback:
- contaners is spelt wrong
- To match all the other labels, it should be `kompose.init.container.name` and `kompose.init.container.image` and `kompose.init.container.command`.

Good addition! Again, apologizes for the delay.

cdrage commented on pkg/loader/compose/utils.go:
```diff
@@ -87,6 +87,12 @@ const (
+	// LabelInitContainerImage defines image to pull
+	LabelInitContainerImage = "kompose.init-contaners.image"
+	// LabelInitContainerComand defines comands
+	LabelInitContainerComand = "kompose.init-contaners.command"
```
LabelInitContainerComand should be spelled LabelInitContainerCommand

sosan commented:
> Sorry for the delay!
> 
> Some feedback:
> 
> * contaners is spelt wrong
> * To match all the other labels, it should be `kompose.init.container.name` and `kompose.init.container.image` and `kompose.init.container.command`.
> 
> Good addition! Again, apologizes for the delay.

changed labels to kompose.init.container.name, kompose.init.container.image, kompose.init.container.command
fixed documentation, tests and e2e
also merged main with new commits to this branch

cdrage commented:
/lgtm
